arangod/IResearch/IResearchDocument.cpp

```diff
@@ -583,14 +614,14 @@ bool FieldIterator::setRegularAttribute(IResearchLinkMeta const& context) {
     case VPackValueType::Int:
     case VPackValueType::UInt:
     case VPackValueType::SmallInt:
-      setNumericValue(value, nameBuffer(), _value);
+      setNumericValue(value);
```

Comment:
this should probably use a different method than the method called for double values, as integers may lose precision when being converted into a double.

Reply:
see above